### ISSUE TITLE
F-Droid build failed

### ISSUE THREAD

**Issue by linsui** [OMGSoundboard/android-app#109]
## :writing_hand: Describe the bug
https://gitlab.com/fdroid/fdroiddata/-/jobs/8017535739 The wear os deps are not FOSS. Could you please provide a FOSS flavor for F-Droid? Thanks!

## :iphone: Tech info
- Device: 
- OS: 
- App Version:

**Comment by OMGSoundboard** [comment]
Thanks for reporting the issue, @linsui.
May you simply _ignore_ the whole `wear` folder?

**Comment by licaon-kter** [comment]
Isn't that tied in the main code?

**Comment by OMGSoundboard** [comment]
> Isn't that tied in the main code?

No, the smartwatch Wear OS by Google part is independent from the smartphone part.
Am I wrong, @Rashxz?

**Comment by linsui** [comment]
It's not that case anymore. https://github.com/OMGSoundboard/android-app/pull/100 added wear related features in the main app.

**Comment by Rashxz** [comment]
> > Isn't that tied in the main code?
> 
> No, the smartwatch Wear OS by Google part is independent from the smartphone part. Am I wrong, @Rashxz?

The main app also has the wear play services library to comunicate and sync data. I'll make a flavor to leave that part out.

**Comment by OMGSoundboard** [comment]
May you please try if `v13.0` builds again on F-Droid, @linsui?

**Comment by linsui** [comment]
https://github.com/OMGSoundboard/android-app/blob/v13.0/core/build.gradle#L48 This is not FOSS. Could you please remove this from foss flavor?

**Comment by Rashxz** [comment]
> https://github.com/OMGSoundboard/android-app/blob/v13.0/core/build.gradle#L48 This is not FOSS. Could you please remove this from foss flavor?

Forgot that was there sorry. Removed.

**Comment by linsui** [comment]
https://github.com/OMGSoundboard/android-app/blob/v13.0/sync_manager/build.gradle#L54 This lib depends on GMS.

**Comment by Rashxz** [comment]
> https://github.com/OMGSoundboard/android-app/blob/v13.0/sync_manager/build.gradle#L54 This lib depends on GMS.

moved, do you see any other lib that is not foss?

**Comment by linsui** [comment]
No, it should be good now. :)

**Comment by OMGSoundboard** [comment]
I created a new release and hope it builds now. 🫡

**Comment by linsui** [comment]
https://github.com/OMGSoundboard/android-app/blob/development/core/build.gradle#L44 I didn't notice this, sorry... Could you please also remove it?

**Comment by Rashxz** [comment]
Removed

**Comment by OMGSoundboard** [comment]
I created a new release and hope it builds now. 😀

**Comment by licaon-kter** [comment]
still fails: https://gitlab.com/fdroid/fdroiddata/-/jobs/8342676205#L658

diff log: [omg17.log](https://github.com/user-attachments/files/17729716/omg17.log)

**Comment by IzzySoft** [comment]
v17.0 fails RB at IzzyOnDroid as well btw, with a huge diff in `classes2.dex`: [diffs.zip](https://github.com/user-attachments/files/17737492/diffs.zip)

Do you still build with JDK-17 as your workflow states? Well, results with JDK-21 were identical, so that's not it. Built from a clean tree – or any local changes/artifacts/caches?

**Comment by OMGSoundboard** [comment]
I suppose @Rashxz is already looking into it. I have no clue at the moment, sorry.

**Comment by IzzySoft** [comment]
Fingers crossed then! As discussed on "the other channel", v17.0 is marked "RB failed" now (afraid we cannot "rescue" that one) – but that "mark" will age out with future releases hopefully matching up again :crossed_fingers:

**Comment by linsui** [comment]
Kindly ping. :) Any update here?

**Comment by OMGSoundboard** [comment]
Sorry for the delay, @linsui.
We can try a new build `v18.0` next month.

**Comment by OMGSoundboard** [comment]
This month was too busy so the next release is possible next month at the earliest. 🙈

**Comment by linsui** [comment]
Kindly ping. :)

**Comment by OMGSoundboard** [comment]
> Kindly ping. :)

Sorry for the delay, @linsui. We will try it next month.

**Comment by OMGSoundboard** [comment]
Sorry again, @linsui & @IzzySoft.

Can you build this from `trunk` too?

[OMGSoundboard_Foss_18.zip](https://github.com/user-attachments/files/21638183/OMGSoundboard_Foss_18.zip)
[OMGSoundboard_Google_18.zip](https://github.com/user-attachments/files/21638289/OMGSoundboard_Google_18.zip)
[OMGSoundboard_Google_18_APK.zip](https://github.com/user-attachments/files/21638290/OMGSoundboard_Google_18_APK.zip)
[OMGSoundboard_Wear_18.zip](https://github.com/user-attachments/files/21638288/OMGSoundboard_Wear_18.zip)
[OMGSoundboard_Wear_18_APK.zip](https://github.com/user-attachments/files/21638287/OMGSoundboard_Wear_18_APK.zip)

**Comment by IzzySoft** [comment]
The Foss one is RB here.

**Comment by licaon-kter** [comment]
FOSS confirmed for F-Droid too 👍 

@OMGSoundboard ping us when a release is ready

**Comment by OMGSoundboard** [comment]
Thanks everyone! 😍
The new release is ready, @licaon-kter & @IzzySoft.

**Comment by IzzySoft** [comment]
It already passed our second builder, but missed the repo updater by just 15 minutes 🙈 Will be part of the next sync then.

**Comment by linsui** [comment]
Thanks, unfortunately, AGP 8.12.0 doesn't work on our old buildserver. Could you please downgrade it?

**Comment by Rashxz** [comment]
> Thanks, unfortunately, AGP 8.12.0 doesn't work on our old buildserver. Could you please downgrade it?

Sure, whats the latest version your buildserver supports?

**Comment by licaon-kter** [comment]
8.12.0-alpha07 appears to be the latest one compatible

**Comment by linsui** [comment]
https://gitlab.com/fdroid/fdroiddata/-/jobs/12059374122 Reproducible build failed.

**Comment by IzzySoft** [comment]
related: https://github.com/OMGSoundboard/android-app/issues/190

in short:

* @linsui skip 19, next release should be fine again).
* @Rashxz that AGP version limitation should not apply anymore, if I got that right, as F-Droid now has a new build machine.

**Comment by licaon-kter** [comment]
> that AGP version limitation should not apply anymore, if I got that right, as F-Droid now has a new build machine.

true

**Comment by IzzySoft** [comment]
TWIF reader here 😉

**Comment by OMGSoundboard** [comment]
New release is live. Does it build fine for both of you, @licaon-kter & @IzzySoft?

**Comment by IzzySoft** [comment]
Confirmed here, thanks!

**Comment by licaon-kter** [comment]
Here too 👍
